elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/ElasticsearchHttpClientConfigurer.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.client;
+
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+
+/**
+ * @author Yoann Rodiere
+ * @hsearch.experimental This contract is currently under active development and may be altered in future releases,
+ */
+public interface ElasticsearchHttpClientConfigurer {
+
+	HttpAsyncClientBuilder configure(HttpAsyncClientBuilder builder);
```

Comment:
> It's not clear to me why the method doesn't return void ?
> I don't think we should allow users to return a different instance.

I generally take care to always use the latest builder returned from the builder methods, but it's just to be safe: the builder could, in theory, choose to return a different instance at some point. In which case we need the users to return the latest instance.
This builder doesn't do that currently, so if you think it's unlikely it ever will, then yes, we can return void instead.  

> This API (SPI?) will need some proper javadoc.

True, sorry about that :x I will add it.